File: app/src/main/java/com/example/benedict/internetconnection/WebAppInterface.java
package com.example.benedict.internetconnection;

import android.content.Context;
import android.webkit.JavascriptInterface;
import android.widget.Toast;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

public class WebAppInterface {

    public static String TAG = WebAppInterface.class.getSimpleName();
    private Context context;
    private MutableLiveData<String> liveText;

    WebAppInterface(Context context) {
        this.context = context;
        this.liveText = new MutableLiveData<>();
    }

    @JavascriptInterface // Show a toast from the web page
    public void showToast(String toast) {
        Toast.makeText(context, toast, Toast.LENGTH_SHORT).show();
    }

    @JavascriptInterface
    public int getAndroidVersion() {
        return android.os.Build.VERSION.SDK_INT;
    }

    @JavascriptInterface
    public void showToastAndroidVersion(String versionName) {
        Toast.makeText(context, versionName, Toast.LENGTH_SHORT).show();
    }

    @JavascriptInterface
    public void postLiveText(String value) {
        liveText.postValue(value);
    }

    public LiveData<String> observeLiveText() {
        return liveText;
    }

    public String getJavascriptAlert(String alert) {
        return "javascript:alert(showWebAlert('" + alert + "'))";
    }

    public String getJavascriptAndroidVersion(String paragraph) {
        return "javascript:alert(showAndroidVersion('" + paragraph + "'))";
    }

    public String getJavascriptParagraph(String paragraph) {
        return "javascript:setParagraph(\"" + paragraph + "\")";
    }
}//javascript:setParagraph

File: app/src/main/java/com/example/benedict/webviewconnection/WebAppInterface.kt
package com.example.benedict.webviewconnection

import android.content.Context
import android.os.Build
import android.webkit.JavascriptInterface
import android.widget.Toast
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData

public class WebAppInterface {

    companion object {
        public val TAG = WebAppInterface::class.java.getSimpleName()
    }

    private val context : Context
    private val liveText : MutableLiveData<String>

    constructor(context : Context) {
        this.context = context
        this.liveText = MutableLiveData<String>()
    }

    @JavascriptInterface // Show a toast from the web page
    public fun showToast(toast : String) {
        Toast.makeText(context, toast, Toast.LENGTH_SHORT).show()
    }

    @JavascriptInterface
    public fun getAndroidVersion() : Int {
        return Build.VERSION.SDK_INT
    }

    @JavascriptInterface
    public fun showToastAndroidVersion(versionName : String) {
        Toast.makeText(context, versionName, Toast.LENGTH_SHORT).show()
    }

    @JavascriptInterface
    public fun postLiveText(value : String) {
        liveText.postValue(value)
    }

    public fun observeLiveText() : LiveData<String> {
        return liveText
    }

    public fun getJavascriptAlert(alert : String) : String {
        return "javascript:alert(showWebAlert('$alert'))"
    }

    public fun getJavascriptAndroidVersion(paragraph : String) : String {
        return "javascript:alert(showAndroidVersion('$paragraph'))"
    }

    public fun getJavascriptParagraph(paragraph : String) : String {
        return "javascript:setParagraph(\"$paragraph\")"
    }
}